network/network.go

```diff
@@ -188,4 +196,9 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 			host,
 		)
 	}
+
+	t := time.Now()
+	elapsed := t.Sub(start)
+	c.logger.Log("level", "info", "message", fmt.Sprintf("Scrap finished, took %#q", elapsed))
```

Comment:
This would make more sense a metric, no?

Reply:
The point in the pm is that scrappe took to long so we did not have any metric so we got paged by target down.

The idea was to at least have some info in logs pods saying if that completed in a long/proper time and then know if it was prom that was not able to reach the pod, oveloaded node processing the collect and timeouting...etc 

Does it makes sense?

Reply:
I would rather add `collecting metrics` & `collected metrics` logs at the top and the bottom of the function. As we were talking it would be good to have some `scrapeid` in here to be able to filter and correlate logs for a single Collect call.

Reply:
A metric is already in place in `scrape_duration_seconds{app="net-exporter"}`
I feel we can improve here the pod logs

Added  `collecting metrics` & `collected metrics`to be clearer with an identifier (counter+pod name would identify any log)

Logs would look like this:
```
{"caller":"github.com/giantswarm/net-exporter/vendor/github.com/giantswarm/operatorkit/client/k8srestconfig/k8s_rest_config.go:148","level":"debug","message":"creating in-cluster REST config","time":"2019-09-23T11:48:21.976658+00:00"}
{"caller":"github.com/giantswarm/net-exporter/vendor/github.com/giantswarm/operatorkit/client/k8srestconfig/k8s_rest_config.go:155","level":"debug","message":"created in-cluster REST config","time":"2019-09-23T11:48:21.976833+00:00"}
{"caller":"github.com/giantswarm/net-exporter/vendor/github.com/giantswarm/microkit/server/server.go:308","level":"debug","message":"running server at http://0.0.0.0:8000","time":"2019-09-23T11:48:21.977983+00:00"}
{"caller":"github.com/giantswarm/net-exporter/vendor/github.com/giantswarm/microkit/server/server.go:293","level":"debug","message":"running debug server at http://127.0.0.1:6060/debug","time":"2019-09-23T11:48:21.978169+00:00"}
{"1","caller":"github.com/giantswarm/net-exporter/network/network.go:142","level":"info","message":"collecting metrics 1 ","time":"2019-09-23T11:48:50.284622+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:151","level":"info","message":"collected `net-exporter` service","time":"2019-09-23T11:48:50.29708+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:154","level":"info","message":"connecting to kubernetes api to get `net-exporter` endpoints","time":"2019-09-23T11:48:50.297113+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:162","level":"info","message":"collected endpoints in `net-exporter`","time":"2019-09-23T11:48:50.304642+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:188","level":"info","message":"dialing host `192.168.66.188:8000` took 0.00019631s","time":"2019-09-23T11:48:50.304942+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:188","level":"info","message":"dialing host `192.168.161.171:8000` took 0.000552709s","time":"2019-09-23T11:48:50.305506+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:188","level":"info","message":"dialing host `192.168.209.136:8000` took 0.000747722s","time":"2019-09-23T11:48:50.305739+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:188","level":"info","message":"dialing host `192.168.49.28:8000` took 0.000759224s","time":"2019-09-23T11:48:50.30588+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:181","host":"192.168.171.7:8000","level":"error","message":"could not dial host","stack":"dial tcp 192.168.171.7:8000: i/o timeout","time":"2019-09-23T11:48:55.305013+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:181","host":"172.31.132.48:8000","level":"error","message":"could not dial host","stack":"dial tcp 172.31.132.48:8000: i/o timeout","time":"2019-09-23T11:48:55.305013+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:207","level":"info","message":"collected metrics 1, took 5.02067057s","time":"2019-09-23T11:48:55.305179+00:00"}
```